pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -0,0 +1,312 @@
+	}
+	// Success! Let's update the status
+	if cephCluster.Spec.External.Enable {
+		r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionConnected)
+	} else {
+		r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionReady)
+	}
+
+	// Return and do not requeue
+	logger.Debug("done reconciling")
+	return reconcile.Result{}, nil
+}
+
+// Create the ceph blockpool rados namespace
+func (r *ReconcileCephBlockPoolRadosNamespace) createOrUpdateRadosNamespace(cephBlockPoolRadosNamespace *cephv1.CephBlockPoolRadosNamespace) error {
+	logger.Infof("creating ceph blockpool rados namespace %s in namespace %s", cephBlockPoolRadosNamespace.Name, cephBlockPoolRadosNamespace.Namespace)
+
+	err := cephclient.CreateRadosNamespace(r.context, r.clusterInfo, cephBlockPoolRadosNamespace.Spec.BlockPoolName, cephBlockPoolRadosNamespace.Name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create ceph blockpool rados namespace %q", cephBlockPoolRadosNamespace.Name)
+	}
+
+	return nil
+}
+
+// Delete the ceph blockpool rados namespace
+func (r *ReconcileCephBlockPoolRadosNamespace) deleteRadosNamespace(cephBlockPoolRadosNamespace *cephv1.CephBlockPoolRadosNamespace) error {
+	logger.Infof("deleting ceph blockpool rados namespace object %q", cephBlockPoolRadosNamespace.Name)
+	if err := cephclient.DeleteRadosNamespace(r.context, r.clusterInfo, cephBlockPoolRadosNamespace.Spec.BlockPoolName, cephBlockPoolRadosNamespace.Name); err != nil {
+		return errors.Wrapf(err, "failed to delete ceph blockpool rados namespace %q", cephBlockPoolRadosNamespace.Name)
+	}
+
+	logger.Infof("deleted ceph blockpool rados namespace %q", cephBlockPoolRadosNamespace.Name)
+	return nil
+}
+
+// updateStatus updates an object with a given status
+func (r *ReconcileCephBlockPoolRadosNamespace) updateStatus(client client.Client, name types.NamespacedName, status cephv1.ConditionType) {
+	cephBlockPoolRadosNamespace := &cephv1.CephBlockPoolRadosNamespace{}
+	if err := client.Get(r.opManagerContext, name, cephBlockPoolRadosNamespace); err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("CephBlockPoolRadosNamespace resource not found. Ignoring since object must be deleted.")
+			return
+		}
+		logger.Warningf("failed to retrieve ceph blockpool rados namespace %q to update status to %q. %v", name, status, err)
+		return
+	}
+	if cephBlockPoolRadosNamespace.Status == nil {
+		cephBlockPoolRadosNamespace.Status = &cephv1.CephBlockPoolRadosNamespaceStatus{}
+	}
+
+	cephBlockPoolRadosNamespace.Status.Phase = status
+	cephBlockPoolRadosNamespace.Status.Info = map[string]string{"clusterID": buildClusterID(cephBlockPoolRadosNamespace)}
+	if err := reporting.UpdateStatus(client, cephBlockPoolRadosNamespace); err != nil {
+		logger.Errorf("failed to set ceph blockpool rados namespace %q status to %q. %v", name, status, err)
+		return
+	}
+	logger.Debugf("ceph blockpool rados namespace %q status updated to %q", name, status)
```

Comment:
The rook operator can operate on multiple resources in multiple namespaces. All info/debug messages must list the namespaced name so context is clear.